lib/lexer.mli

```diff
@@ -1 +1,23 @@
-val hello_world : unit -> string
+(** The [Lexer] module provides functions for tokenizing (a subset of) TypeScript source code.
+
+    It defines a lexer type [t] and functions for creating a lexer and consuming tokens.
+
+    The lexer is designed to be immutable: consuming a token returns a new lexer and the
+    consumed token. When there are no more tokens, the lexer returns [Token.EOF]. *)
+
+type t
+
+val of_string : string -> t
```

Comment:
So these are functions we make available to the "outside world" and `lexer.ml` will house the logic? Am I correct in that?

Reply:
Yep, exactly. The project won't compile until these signatures have corresponding implementations in the `lexer.ml` file.

(The `lexer.ml` can create as many functions as it needs though, it just at minimum needs to satisfy all the signatures with an implementation)

Reply:
I like it - almost too much...